.github/workflows/test.yml

```diff
@@ -0,0 +1,100 @@
+name: Run test on linux mac
+
+on:
+  push:
+    branches: [ master ]
+    tags: [ '*' ]
+  pull_request:
+
+env:
+  BUILD_DIR: build
+  BUILD_TREADS: 2
+  CTEST_OUTPUT_ON_FAILURE: 1
+  GITHUB_HUNTER_USERNAME: ${{ secrets.HUNTER_USERNAME }}
+  GITHUB_HUNTER_TOKEN: ${{ secrets.HUNTER_TOKEN }}
+  INDOCKER_IMAGE: soramitsu/kagome-dev:8
+
+jobs:
+  MacOS:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: install
+      run: ./housekeeping/macos/dependency.sh
+    - name: build
+      env:
+        BUILD_TREADS: 4
+        DEVELOPER_DIR: /Applications/Xcode_11.app/Contents/Developer
+      run: ./housekeeping/makeBuild.sh  -DCOVERAGE=OFF -DCMAKE_TOOLCHAIN_FILE=cmake/toolchain/cxx17.cmake
+
+  Linux:
+      strategy:
+        matrix:
+          options:
+           - name: "Linux: gcc-9 ASAN No Toolchain"
+             run: ./housekeeping/makeBuild.sh -DCMAKE_TOOLCHAIN_FILE=cmake/toolchain/gcc-9_cxx17.cmake -DASAN=ON
+           - name: "Linux: clang-8 TSAN"
+             run: ./housekeeping/makeBuild.sh -DCMAKE_TOOLCHAIN_FILE=cmake/san/clang-8_cxx17_tsan.cmake
+           - name: "Linux: clang-8 UBSAN"
+             run: ./housekeeping/makeBuild.sh -DCMAKE_TOOLCHAIN_FILE=cmake/san/clang-8_cxx17_ubsan.cmake
+      name: "${{ matrix.options.name }}"
+      runs-on: ubuntu-latest
+      container: soramitsu/kagome-dev:8 # Change to ${{ env.INDOCKER_IMAGE }} https://github.community/t/how-to-set-and-access-a-workflow-variable/17335/6
+      steps:
+        - uses: actions/checkout@v2
+        - name: "${{ matrix.options.name }}"
+          run: "${{ matrix.options.run }}"
+
+  clang-tidy:
+      name: "Linux: clang-tidy"
+      runs-on: ubuntu-latest
+      container: soramitsu/kagome-dev:8
+      steps:
+        - uses: actions/checkout@v2
+        - name: clang-tidy
+          env:
+            # build only generated files, so clang-tidy will work correctl
+            BUILD_TARGET: generated
+          run: |
+            ./housekeeping/makeBuild.sh
+            ./housekeeping/clang-tidy.sh ${BUILD_DIR}
+  coverage:
+      name: "Linux: gcc-8 coverage/sonar"
+      runs-on: ubuntu-latest
+      container: soramitsu/kagome-dev:8  # Change to ${{ env.INDOCKER_IMAGE }} https://github.community/t/how-to-set-and-access-a-workflow-variable/17335/6
+      steps:
+        - uses: actions/checkout@v2
+        - name: makeBuild
+          env:
+            BUILD_TARGET: ctest_coverage
+          run: ./housekeeping/makeBuild.sh -DCMAKE_TOOLCHAIN_FILE=cmake/toolchain/gcc-8_cxx17.cmake -DCOVERAGE=ON
+        - name: Submit Coverage
+          env:
+            CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+          run: ./housekeeping/codecov.sh ${BUILD_DIR} ${CODECOV_TOKEN}
```

Comment:
Why do we need to set 'BUILD_DIR' as a script parameter? Other scripts don't need it.

Reply:
This script already presented in repository, I decided  not to change it, because someone may use it.
I will add support of environment variable  to script.

Reply:
fixed